llvm/lib/Target/EraVM/MCTargetDesc/EraVMMCCodeEmitter.cpp

```diff
@@ -91,7 +99,26 @@ uint64_t EraVMMCCodeEmitter::getMemOpValue(const MCInst &MI, unsigned Idx,
 uint64_t EraVMMCCodeEmitter::getCCOpValue(const MCInst &MI, unsigned Idx,
                                           SmallVectorImpl<MCFixup> &Fixups,
                                           const MCSubtargetInfo &STI) const {
-  llvm_unreachable("Unable to encode predicate operand!");
+  switch (MI.getOperand(Idx).getImm()) {
+  default:
+    llvm_unreachable("Invalid condition code");
+  case EraVMCC::COND_INVALID:
+    llvm_unreachable("Cannot encode COND_INVALID");
+  case EraVMCC::COND_NONE:
```

Comment:
For non-invalid, not none members of the enum, can't we just static cast to `uint64_t` and if necessary reenumerate the enum?

Reply:
Addressed via 388ead4c8c9cc83158551e3422570f8f382da955